File: mvpboilerplate/src/main/java/com/arsldev/lutluthfi/mvpboilerplate/base/PlateBasePresenter.java
package com.arsldev.lutluthfi.mvpboilerplate.base;

public class PlateBasePresenter<V extends IPlateBaseView> implements IPlateBasePresenter<V> {

    private V mView;

    @Override
    public void onAttach(V view) {
        this.mView = view;
    }

    @Override
    public void onDetach() {
        this.mView = null;
    }

    protected V getView() {
        return mView;
    }
}

File: mvpboilerplate/src/main/java/com/lutluthfi/mvpboilerplate/base/PlateBasePresenter.kt
package com.lutluthfi.mvpboilerplate.base

open class PlateBasePresenter<V : IPlateBaseView>(view: V) : IPlateBasePresenter<V> {

    private var isAttached: Boolean = false
    protected var view: V? = null
        private set

    init {
        this.view = view
    }

    override fun onAttach() {
        this.isAttached = true
    }

    override fun onDetach() {
        this.view = null
        this.isAttached = false
    }

    override fun isAttached(): Boolean {
        return isAttached
    }
}
